src/client/connection.rs

```diff
@@ -62,42 +51,34 @@ impl Debug for Connection {
     }
 }
 
-pub(crate) struct ConnectOpts {
-    pub encryption: EncryptionLevel,
-    pub trust_cert: bool,
-    pub auth: AuthMethod,
-    pub database: Option<String>,
-    pub instance_name: Option<String>,
-}
-
 enum LoginResult {
     Ok,
     #[cfg(windows)]
     Windows(Box<dyn NextBytes>),
 }
 
-impl Connection {
+impl<S: futures::AsyncRead + futures::AsyncWrite + Unpin + Send> Connection<S> {
```

Comment:
You might want to use futures::{AsyncRead, AsyncWrite} to save valuable horizontal space.